src/metabase/driver/sql/parameters/substitution.clj

```diff
@@ -243,8 +233,39 @@
         (sql.qp/date driver :day identifier)
         identifier)))))
 
+(s/defn ^:private field-filter->replacement-snippet-info :- ParamSnippetInfo
+  "Return `[replacement-snippet & prepared-statement-args]` appropriate for a field filter parameter."
+  [driver {{param-type :type, value :value :as params} :value field :field :as _field-filter}]
+  (let [prepend-field
+        (fn [x]
+          (update x :replacement-snippet
+                  (partial str (field->identifier driver field param-type) " ")))]
+    (cond
+      (ops/operator? param-type)
+      (let [[snippet & args]
+            (->> (ops/to-clause (assoc params :target
+                                       [:template-tag [:field (:name field)
+                                                       {:base-type (:base_type field)}]]))
+                 (mbql.u/desugar-filter-clause)
+                 (wrap-value-literals/wrap-value-literals-in-mbql)
+                 (sql.qp/->honeysql driver/*driver*)
+                 (hsql/format-predicate))]
+        {:replacement-snippet snippet, :prepared-statement-args (vec args)})
```

Comment:
This i think should be the way the rest of these work. MBQL is quite nice to target (as evidenced by the essentially single line that accomplishes all of the `:string/` and `:number/` operators. Targeting this and then compiling mbql to the target dialect seems far preferable than hand-rolling another sql compiler here.

Certainly with emitting mongo i benefited from the knowledge in that rather than creating my own. And perhaps the hand optimized code might be a bit faster, but that's unproven, and a good motivator to optimize mbql in general to get this speedup everywhere.

Reply:
Agreed. We should refactor the old stuff to use the new pattern (either as part of this PR or as a follow-on)